proto/content_analysis/sdk/analysis.proto

```diff
@@ -97,5 +109,8 @@ message ContentAnalysisResponse {
   }
   repeated Result results = 4;
 
+  // Optional handshake message. If this is set, other optional fields should not be set.
+  optional Handshake handshake = 200;
```

Comment:
I think I have a better idea after looking at this PR how the ack and HS messages are used.  Since ContentAnalysisRequest and ContentAnalysisResponse are shared the online CAC, I wonder if we should take a different approach.

Neither ack nor HS are actual types of content analysis requests and responses, they are really different messages altogether.  It just so happens the current LCAC code assumes the only messages send between chrome and the agent are content analysis requests and responses.  I think it would be to define a top level LCAL requst and response, and then embed the other messages within.  For example, something like:
```
message AgentRequest {
  optional ContentAnalysisRequest analysis_request = 1;
  optional Handshake handshake = 2;
  optional Acknowledgement acknowledgement = 3;
}

message AgentResponse {
  optional ContentAnalysisResponse analysis_response = 1;
  optional Handshake handshake = 2;
}
```

Reply:
This is a nice approach! Would this have any affect the Agent's public API?

Reply:
I think the public API should remain as is.  The AgentRequest|Response messages really are internal details of the SDK.  Does that make sense?

Reply:
Yeah for sure!

Reply:
I've started an implementation of this and overall it feels less easy to work with.

Particularly around parsing/copying/passing messages, it turns into something like this:

```
AgentResponse agent_response;
ContentAnalysisResponse analysis_response;

if (!agent_response.ParseFromArray(buffer.data(), buffer.size())) {
  return -1;
}
if (!agent_response.has_analyisis_request()) {
  return -1;
}
analysis_response.CopyFrom(agent_response.analysis_response());
```

Whereas the current implementation looks as follows:

```
ContentAnalysisResponse response;

if (!response.ParseFromArray(buffer.data(), buffer.size())) {
  return -1;
}
```

Reply:
A similar example for sending a proto. A lot of extra code required (the whole mutable_analysis_response thing).

```
AgentResponse agent_response;
ContentAnalysisResponse analysis_response;
std::string response_str;

agent_response.mutable_analysis_response()->CopyFrom(analysis_response);

if (!agent_response.SerializeToString(response_str)) {
  return -1;
}

WriteFile(...)
```

Compared to the current implementation:

```
ContentAnalysisResponse response;
std::string response_str;

if (!response.SerializeToString(response_str)) {
  return -1;
}

WriteFile(...)
```